neo/Consensus/ConsensusService.cs

```diff
@@ -502,10 +502,14 @@ private void OnRegeneration(ConsensusPayload payload, Regeneration message)
             /// </summary>
             if (nValidSignatures >= context.M)
             {
-                Log($"{nameof(OnRegeneration)}: Sorry. I lost some part of the history. I give up...");
+                Log($"{nameof(OnRegeneration)}: Sorry. I lost some part of the history. I give up... Thanks index={payload.ValidatorIndex}");
                 InitializeConsensus(message.ViewNumber);
```

Comment:
Hi @wanglongfei88,
Thanks for all the help given here.

This was recently a tip from @erikzhang. If the Regeneration Payload has a view like 5 we will need that the node also stay in that view for verifying Payloads accordingly.
Maybe the Reset would return it to the first view and, sometimes, we would need it to wait until timer synchronization.

Reply:
Let's think about it.

Change view stuck are still a problem that I am getting here, but it is normal, because I am testing blocks with 1s, then, the formula works just exponentially 2,4,8,16,32.

But I think some payload are not arriving when views are different like this.

Reply:
@wanglongfei88,

I think that the change view should be like you suggested before:
`if (message.ViewNumber != context.ViewNumber && message.Type != ConsensusMessageType.ChangeView && message.Type != ConsensusMessageType.Regeneration)`

I first tried like these but then I updated to an || that I probably destroyed the messages of OnRegeneration and OnChangeView for higher views.

Reply:
You are right, @vncoelho.